rootfs/api/models/app.py

```diff
@@ -555,8 +555,18 @@ def deploy(self, release, force_deploy=False):
                 ) for scale_type, kwargs in deploys.items()
             ]
 
-            async_run(tasks)
+            try:
+                async_run(tasks)
+            except KubeException as e:
+                err = 'There was a problem deploying {}. Rolling back process types to release {}.'.format(version, "v{}".format(release.previous().version))  # noqa
+                # This goes in the log before the rollback starts
+                self.log(err, logging.ERROR)
+                # revert all process types to old release
+                self.deploy(release.previous(), force_deploy=True)
```

Comment:
It will recurse all the way until it finds a good deploy, and the last one would be `no build` which we bail out on at the top, or you hit the 100 recursion limit (or so) in python.